src/controllers/AdminController.ts

```diff
@@ -162,9 +162,49 @@ export class AdminAPI extends Controller {
 
     this.setStatus(201);
 
-    return template;
+    return template
   }
 
+  /**
+   * Create a templates. An overview of Template usage in Retraced can be found at
+   *
+   * https://boxyhq.com/docs/retraced/advanced/display-templates
+   *
+   * @param auth          Base64 encoded JWT
+   * @param projectId     The project id
+   * @param environmentId The environment id
+   * @param body          The template resource to create
+   */
+    @Post("project/{projectId}/templates/bulk")
+    @SuccessResponse("201", "Created")
+    public async createTemplates(
+      @Header("Authorization") auth: string,
+      @Path("projectId") projectId: string,
+      @Query("environment_id") environmentId: string,
+      @Body() body: {templates: TemplateValues[]},
+      @Request() req: express.Request
+    ): Promise<TemplateResponse[]> {
+      if (body.templates.length > 1000) {
+        throw { status: 400, err: new Error("Can only create 100 templates at once.") };
+      }
+
+      await audit(req, "template.create.many", "c", {
+        target: {
+          fields: body,
+        },
+      });
+
+      const templates: TemplateResponse[] = []
+      body.templates.map(async (templateToCreate) => {
+        const template = await createTemplate(auth, projectId, environmentId, Object.assign(templateToCreate, { id: uniqueId() }));
+        templates.push(template)
+      })
```

Comment:
```suggestion
      const templates: TemplateResponse[] = await Promise.all(
      body.templates.map(async (templateToCreate) => {
        const template = await createTemplate(
          auth,
          projectId,
          environmentId,
          Object.assign(templateToCreate, { id: uniqueId() })
        );
        return template;
      })
    );
```

Reply:
Also the function `checkAdminAccessUnwrapped` is getting called inside `createTemplate`, it is required to be called only once per API call but with this structure it will be called per template, can we optimise that

Reply:
I moved the call out of createTemplate only when creating in bulk, i noticed `checkAdminAccessUnwrapped` is always called within the action's function. Let me know if theres a better way to do this.